tests/tfork.c

```diff
@@ -0,0 +1,111 @@
+/* Copyright (C) 2022 Simo Sorce <simo@redhat.com>
+   SPDX-License-Identifier: Apache-2.0 */
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <openssl/evp.h>
+#include <openssl/store.h>
+#include <openssl/rand.h>
+
+static EVP_PKEY *gen_key(void)
+{
+    char label[] = "##########: Fork Test Key";
+    unsigned char id[16];
+    size_t rsa_bits = 3072;
+    OSSL_PARAM params[5];
+    EVP_PKEY_CTX *ctx;
+    EVP_PKEY *key = NULL;
+    int miniid;
+    int ret;
+
+    /* RSA */
+    ret = RAND_bytes(id, 16);
+    if (ret != 1) {
+        fprintf(stderr, "Failed to set generate key id\n");
+        exit(EXIT_FAILURE);
```

Comment:
Could you please add here and to other similar places `ERR_print_error_fp(stderr)`?
It will also require `#include <openssl/err.h>`

Reply:
Ah printing the OpenSSL errors?

I guess I can yeah.